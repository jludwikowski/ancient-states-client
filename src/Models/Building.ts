import BuildingPrototype from './BuildingPrototype';

export default interface Building extends BuildingPrototype{
    finalCost: number,
    finalBuildingTime: number,
    level: number
}
