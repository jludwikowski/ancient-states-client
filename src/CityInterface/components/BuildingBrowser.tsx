import React from "react";
import Building from "../../Models/Building";
import { BuildingCard } from "../components/BuildingCard";

type Props = {
    buildingList: Building[],
    onBuild(id: number): void;
};

export const BuildingBrowser: React.FunctionComponent<Props> = ({buildingList, onBuild}) => (
    <div className="card-group">
        {buildingList.map((building, index) => (
            <React.Fragment key={building.id}>
                <BuildingCard building={building} onBuild={onBuild}/>
            </React.Fragment>
        ))}
    </div>
);
