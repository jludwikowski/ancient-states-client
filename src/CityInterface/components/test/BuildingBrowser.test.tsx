import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import Building from "../../../Models/Building";
import { BuildingBrowser } from '../BuildingBrowser';

describe('BuildingBrowser', () => {

    const buildings:Building[] =  [
            {
                baseBuildingTime: 120,
                baseCost: 100,
                description: 'You recruit standard units here',
                id: 1,
                imageUrl: 'images/icons/icons8-military-base-100.png',
                level: 10,
                name: 'barracks'
            },
            {
                baseBuildingTime: 150,
                baseCost: 250,
                description: 'This building let you upgrade weapons',
                id: 2,
                imageUrl: 'images/icons/icons8-military-base-100.png',
                level: 10,
                name: 'forge'
            },
            {
                baseBuildingTime: 150,
                baseCost: 60,
                description: 'This building produce food',
                id: 3,
                imageUrl: 'images/icons/icons8-military-base-100.png',
                level: 30,
                name: 'fields'
            },
            ];

    const onBuild = (id:number) => {
        console.log('On onDismiss');
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BuildingBrowser buildingList={buildings} onBuild={onBuild} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Should match snapshot', () => {
        const component = shallow(<BuildingBrowser buildingList={buildings} onBuild={onBuild} />);
        expect(component).toMatchSnapshot();
    });
});
