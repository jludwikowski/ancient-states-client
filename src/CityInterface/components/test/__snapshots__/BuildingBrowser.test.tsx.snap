// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildingBrowser Should match snapshot 1`] = `
<div
  className="card-group"
>
  <BuildingCard
    building={
      Object {
        "baseBuildingTime": 120,
        "baseCost": 100,
        "description": "You recruit standard units here",
        "id": 1,
        "imageUrl": "images/icons/icons8-military-base-100.png",
        "level": 10,
        "name": "barracks",
      }
    }
    onBuild={[Function]}
  />
  <BuildingCard
    building={
      Object {
        "baseBuildingTime": 150,
        "baseCost": 250,
        "description": "This building let you upgrade weapons",
        "id": 2,
        "imageUrl": "images/icons/icons8-military-base-100.png",
        "level": 10,
        "name": "forge",
      }
    }
    onBuild={[Function]}
  />
  <BuildingCard
    building={
      Object {
        "baseBuildingTime": 150,
        "baseCost": 60,
        "description": "This building produce food",
        "id": 3,
        "imageUrl": "images/icons/icons8-military-base-100.png",
        "level": 30,
        "name": "fields",
      }
    }
    onBuild={[Function]}
  />
</div>
`;
